impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaOrchestrator.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.testing.kafka.common;
+
+import java.util.Map;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.function.IntPredicate;
+
+import io.kroxylicious.testing.kafka.api.KafkaCluster;
+import io.kroxylicious.testing.kafka.api.TerminationStyle;
+
+/**
+ * Orchestrates operation of a Kafka Cluster. Responsible for:
+ * 1. controlling the order of operations, for example ensuring that non-brokers are stopped first.
+ * 2. connecting the selected driver with the users desired cluster configuration
+ */
+public class KafkaOrchestrator implements KafkaCluster {
+
+    private final KafkaDriver driver;
+    private final KafkaTopology kafkaTopology;
+
+    public KafkaOrchestrator(KafkaClusterConfig config, KafkaDriver driver) {
+        this.driver = driver;
+        kafkaTopology = KafkaTopology.create(driver, config);
+    }
+
+    @Override
+    public void start() {
+        kafkaTopology.startZookeeperIfRequired();
+        kafkaTopology.pureBrokersLast().parallel().forEach(KafkaNode::start);
+        Utils.awaitExpectedBrokerCountInClusterViaTopic(
+                kafkaTopology.getAnonymousClientConfiguration(), 120,
+                TimeUnit.SECONDS,
+                getNumOfBrokers());
+    }
+
+    @Override
+    public int addBroker() {
+        KafkaNode node = kafkaTopology.addBroker();
+        node.start();
+        Utils.awaitExpectedBrokerCountInClusterViaTopic(
+                kafkaTopology.getAnonymousClientConfiguration(), 120,
+                TimeUnit.SECONDS,
+                getNumOfBrokers());
+        return node.nodeId();
+    }
+
+    @Override
+    public void removeBroker(int nodeId) throws UnsupportedOperationException, IllegalArgumentException, IllegalStateException {
+        KafkaNode node = kafkaTopology.get(nodeId);
+        if (node == null) {
+            throw new IllegalArgumentException("nodeId isn't in topology: " + nodeId);
```

Comment:
I realise most of this is actually just shuffling code around but...

I'm wondering if we should let this happen and just return. Why should we prevent people calling remove a nodeId that is already gone? 

I can imagine calling `removeBroker` if `addBroker` threw an exception to ensure it wasn't in some zombie state. 

`node` being `null` here represents the desired state.